core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -197,16 +202,15 @@ public ZenDiscovery(Settings settings, ThreadPool threadPool, TransportService t
         this.masterFD.addListener(new MasterNodeFailureListener());
         this.nodesFD = new NodesFaultDetection(settings, threadPool, transportService, clusterName);
         this.nodesFD.addListener(new NodeFaultDetectionListener());
+        this.pendingStatesQueue = new PendingClusterStatesQueue(logger, settings.getAsInt(SETTINGS_MAX_PENDING_CLUSTER_STATES, 25));
```

Comment:
can we make this a proper setting, so the default is in the object and it gets validated and such? I think this an oversight when the settings were migrated.